ldms/src/ldmsd/ldmsd_decomp.c

```diff
@@ -156,7 +156,7 @@ static ldmsd_decomp_t __decomp_get(const char *decomp, ldmsd_req_ctxt_t reqc)
 		d = dlopen(library_name, RTLD_NOW);
 		if (d)
 			break;
-		if (!stat(library_name, &st))
+		if (stat(library_name, &st))
 			continue;
 		dlerr = dlerror();
 		errno = ELIBBAD;
```

Comment:
If we run through all the pathdir and can't find the decomposer, it doesn't mean the decomposer is bad, it more likely means the specified decomposer does not exist; again the stat() tells you which.

Reply:
This comment is supposed to be at line 169...

Reply:
At line 179, the error "get() not found" is not useful to the user unless they develop decomposers. It would be more useful to say "The library %s is not a valid decomposer plugin."